src/Token.php

```diff
@@ -78,9 +78,24 @@ public function offsetExists($offset): bool
      * @deprecated Use the value, type or position property instead
      * {@inheritDoc}
      *
-     * @param array-key $offset
+     * @param O $offset
      *
      * @return mixed
+     * @psalm-return (
+     *     O is 'value'
+     *     ? string|int
+     *     : (
+     *         O is 'type'
+     *         ? T|null
+     *         : (
+     *             O is 'position'
+     *             ? int
+     *             : mixed
+     *         )
+     *     )
+     * )
```

Comment:
Docs: https://psalm.dev/docs/annotating_code/type_syntax/conditional_types/

I checked that it works with `@psalm-trace` in a test